lib/src/operance_data_controller.dart

```diff
@@ -310,12 +319,32 @@ class OperanceDataController<T> extends ChangeNotifier {
     notifyListeners();
   }
 
-  /// Toggles the expansion of a row.
-  void toggleExpandedRow(int index) {
-    _expandedRows[index] = !(_expandedRows[index] ?? false);
+  /// Toggle the expansion of a row and fetch its data if needed
+  Future<void> toggleExpandedRow(int index) async {
+    final isExpanded = _expandedRows[index] ?? false;
+    _expandedRows[index] = !isExpanded;
+
+    if (!isExpanded && _expandedRowFetcher != null) {
+      if (!_expandedRowData.containsKey(index)) {
+        try {
+          _expandedRowData[index] = Future.value(
+            await _expandedRowFetcher!(allRows[index]),
+          );
+        } on Exception {
+          _expandedRows[index] = false;
+          _expandedRowData.remove(index);
```

Comment:
not sure how to sort slight issue with the error handling. Maybe log the error and notify the users

Reply:
the devs should always handle their own exceptions, we shouldn't handle it for them